src/main/java/de/tum/cit/aet/artemis/core/web/CourseResource.java

```diff
@@ -1453,4 +1455,23 @@ public ResponseEntity<Long> getNumberOfAllowedComplaintsInCourse(@PathVariable L
         long unacceptedComplaints = complaintService.countUnacceptedComplaintsByParticipantAndCourseId(participant, courseId);
         return ResponseEntity.ok(Math.max(complaintService.getMaxComplaintsPerParticipant(course, participant) - unacceptedComplaints, 0));
     }
+
+    @GetMapping("courses/{courseId}/existing-exercise-details")
+    @EnforceAtLeastEditorInCourse
+    public ResponseEntity<CourseExistingExerciseDetails> getExistingExerciseDetails(@PathVariable Long courseId, @RequestParam(defaultValue = "false") Boolean includeShortNames) {
```

Comment:
We currently allow the same name for exercises of different types. Should we add a parameter for the exercise type here to only query the type that is relevant? Otherwise we would have different behavior in simple and hard mode

Reply:
Thanks for the heads up, I assumed the title to be a course-wide identifier, I will adjust the endpoint accordingly 👍